pilot/pkg/model/extensions.go

```diff
@@ -56,6 +92,49 @@ type WasmPluginWrapper struct {
 	WasmExtensionConfig *envoyWasmFilterV3.Wasm
 }
 
+func (p *WasmPluginWrapper) MatchListener(proxyLabels map[string]string, li *WasmPluginListenerInfo) bool {
+	workloadMatch := (p.Selector == nil || labels.Instance(p.Selector.MatchLabels).SubsetOf(proxyLabels))
+	return workloadMatch && matchTrafficSelectors(p.Match, li)
+}
+
+type WasmPluginListenerInfo struct {
+	Port  Port
+	Class istionetworking.ListenerClass
+}
+
+func matchTrafficSelectors(ts []*extensions.WasmPlugin_TrafficSelector, li *WasmPluginListenerInfo) bool {
+	if li == nil || ts == nil {
+		return true
+	}
+
+	for _, match := range ts {
+		if matchMode(match.Mode, li.Class) && matchPorts(match.Ports, li.Port) {
+			return true
+		}
+	}
+	return false
+}
+
+func matchMode(workloadMode typeapi.WorkloadMode, class istionetworking.ListenerClass) bool {
+	if result, ok := workloadMathRule[workloadMode]; ok {
+		return result(class)
+	}
+	return false
+}
+
+func matchPorts(portSelectors []*typeapi.PortSelector, port Port) bool {
+	if len(portSelectors) == 0 {
+		// If there is no specified port, match with all the ports.
+		return true
+	}
+	for _, ps := range portSelectors {
+		if ps.GetNumber() != 0 && ps.GetNumber() == uint32(port.Port) {
```

Comment:
here we only compare the port number, so we do not all other infos in model.Port

Reply:
I just wanted to leave some space for the future extension, but I agree with your comment.
Let's keep it simple.

All the ports in this PR are now using a `int` type instead of `Port struct`.